boot/zephyr/flash_map.c

```diff
@@ -43,44 +43,59 @@ extern struct device *boot_flash_device;
 #define FLASH_MAP_ENTRY_MAGIC 0xd00dbeef
 
 struct flash_map_entry {
-    const uint32_t magic;
-    const struct flash_area area;
+    uint32_t magic;
```

Comment:
These could be kept const if you initialized the array at build time by checking the build configuration and having different static initializers in the two cases.

You could also use a Zephyr SYS_INIT routine to initialize this before main() gets control as a runtime alternative that would still mean you don't need to change APIs shared with Mynewt.

But the const declarations are nice; it'd be better to keep them.